### PR TITLE
 Privatize  AP_VisualOdom

### PR DESCRIPTION
**Pull request by hendjoshsr71**
Privatizes visual odometry logging

### CODE REVIEW

File: libraries/AP_Logger/LogFile.cpp [hunk 1]
```diff
@@ -654,64 +654,6 @@ void AP_Logger::Write_RPM(const AP_RPM &rpm_sensor)
...
-{
-    const struct log_VisualVelocity pkt_visualvel {
-        LOG_PACKET_HEADER_INIT(LOG_VISUALVEL_MSG),
-        time_us         : AP_HAL::micros64(),
-        remote_time_us  : remote_time_us,
-        time_ms         : time_ms,
-        vel_x           : vel.x,
-        vel_y           : vel.y,
-        vel_z           : vel.z,
-        vel_err         : vel_err,
```
Comment (amilcarlucas) [inline libraries/AP_Logger/LogFile.cpp:708]: Why are you droping vel_error?

**Comment by peterbarker** [comment]
@amilcarlucas it still seems to be there?
image

@hendjoshsr71 needs a rebase....

**Comment by hendjoshsr71** [comment]
Rebased... for the moment

File: libraries/AP_Logger/LogFile.cpp (hunk 1 above)
Comment (peterbarker) [reply libraries/AP_Logger/LogFile.cpp:626]: He's not?!

**Review by peterbarker — APPROVED** [review]
LGTM - we've merged others....

**Review by khancyr — APPROVED** [review]

**Review by amilcarlucas — APPROVED** [review]

**Comment by rmackay9** [comment]
@hendjoshsr71 has this been tested?  It's probably fine but we'd like all changes tested. It should be possible to test in SITL and the wiki has instructions on how to run a simulated vicon.  https://ardupilot.org/dev/docs/using-sitl-for-ardupilot-testing.html#testing-vicon-aka-vision-positioning

**Comment by hendjoshsr71** [comment]
@rmackay9  Sorry about that. I should have asked in regards to testing this one. 

I didn't find a way through searching the wiki to simulate the input for the VISO log message. Is there a setting for SITL to generate `mavlink vision_position_delta` messages?

private_VISV_VISP

**Comment by tridge** [comment]
we need something that shows that this has been tested

**Comment by hendjoshsr71** [comment]
Here are before and after graphs for this PR using a SITL mission. Note the VISO messages required additional code to SITL as the Vicon simulation was not sending `MavLink vision_position_delta` messages. I'll PR that separately. 

**Master**
pr_master_VISO_with_sitl_vicon_pos_delta

**After PR**
pr_privateLog_VISO_with_sitl_vicon_pos_delta

**Master**
pr_master_VISP_with_sitl_vicon_pos_delta

**After PR**
pr_privateLog_VISP_with_sitl_vicon_pos_delta

**Master**
pr_master_VISV_with_sitl_vicon_pos_delta

**After PR**
pr_privateLog_VISV_with_sitl_vicon_pos_delta

**Comment by rmackay9** [comment]
Merged, thanks!

**Pull request merged**